### PR TITLE
Fix compiler warnings

### PR DESCRIPTION
**Pull request by ayeshLK**
Fixes [#2392](https://github.com/ballerina-platform/ballerina-standard-library/issues/2392)

## Examples
N/A

## Checklist
- [X] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: ballerina/tests/subscription_initiation_test.bal [hunk 1]
```diff
@@ -26,15 +26,15 @@ const string HUB_FAILURE_URL = "http://127.0.0.1:9192/common/failed";
 const string COMMON_TOPIC = "https://sample.topic.com";
 
 service /common on new http:Listener(9192) {
-    isolated resource function get discovery(http:Caller caller, http:Request request) {
+    isolated resource function get discovery(http:Caller caller, http:Request request) returns error? {
         http:Response response = new;
         response.addHeader("Link", "<http://127.0.0.1:9192/common/hub>; rel=\"hub\"");
         response.addHeader("Link", "<https://sample.topic.com>; rel=\"self\"");
-        http:ListenerError? resp = caller->respond(response);
+        check caller->respond(response);
```
Comment (chamil321) [inline ballerina/tests/subscription_initiation_test.bal:33]: Also you can return

Comment (chamil321) [inline ballerina/tests/subscription_initiation_test.bal:33]:
```suggestion
        return caller->respond(response);
```

File: ballerina/utils.bal [hunk 2]
```diff
@@ -235,8 +231,12 @@ isolated function updateResponseBody(http:Response response, anydata? messageBod
 # 
 # + caller - The `http:Caller` reference for the current request
 # + response - Updated `http:Response`
-isolated function respondToRequest(http:Caller caller, http:Response response) {
+# + return - An `websub:Error` if there is an error while responding to the request or else `()`
+isolated function respondToRequest(http:Caller caller, http:Response response) returns Error? {
     http:ListenerError? responseError = caller->respond(response);
+    if responseError is http:ListenerError {
+        return error Error("Error occurred while responding to the request");
```
Comment (ayeshLK) [inline ballerina/utils.bal:238]:
```suggestion
        return error Error("Error occurred while responding to the request ", responseError);
```

**Review by chamil321 — APPROVED** [review]

File: ballerina/tests/subscription_initiation_test.bal (hunk 1 above)
Comment (ayeshLK) [reply ballerina/tests/subscription_initiation_test.bal:33]: Will do this syntax update in another PR.

**Pull request merged**